src/Orchard.ContentManagement/DefaultContentManager.cs

```diff
@@ -420,6 +428,14 @@ public Task DiscardDraftAsync(ContentItem contentItem)
 
             Handlers.Reverse().Invoke(handler => handler.Removed(context), _logger);
 
+            var publishedItem = GetAsync(contentItem.ContentItemId, VersionOptions.Published).GetAwaiter().GetResult();
```

Comment:
We don't use async when we can "not use async" nd just return the Task. Otehrwise we use it.